tests/chainer_tests/functions_tests/array_tests/test_as_strided.py

```diff
@@ -0,0 +1,322 @@
+import unittest
+
+from chainer import testing, Variable, grad
+
+import numpy as np
+
+from chainer.functions import as_strided
+from chainer.functions.array.as_strided import _stride_array
+
+
+@testing.parameterize(
+    {'dtype': np.float16},
+    {'dtype': np.float32},
+    {'dtype': np.float64},
+    {'dtype': np.int16},
+    {'dtype': np.int32},
+    {'dtype': np.int64}
+)
+class TestStrideArray(unittest.TestCase):
+    def check_flip(self, xp):
+        x = xp.arange(4, dtype=self.dtype)
+        y = _stride_array(x, (4,), (-1,), 3)  # [3, 2, 1, 0]
+        y_expected = xp.flip(x, axis=0)
+        testing.assert_allclose(y, y_expected)
+
+    def test_flip_cpu(self):
+        self.check_flip(np)
+
+    @testing.attr.gpu
+    def test_flip_gpu(self):
+        import cupy as cp
+        self.check_flip(cp)
```

Comment:
please remove `import` in function.
```suggestion
        self.check_flip(cuda.cupy)
```
And please add `from chainer.backends import cuda`.